module/VuFind/src/VuFind/Recommend/ConsortialVuFind.php

```diff
@@ -0,0 +1,242 @@
+ */
+
+namespace VuFind\Recommend;
+
+use Laminas\Config\Config;
+use VuFind\Connection\ConsortialVuFind as Connection;
+
+use function intval;
+
+/**
+ * ConsortialVuFind Recommendations Module
+ *
+ * This class searches a separate instance of VuFind via its public API and links to
+ * the record and results pages hosted within that instance. This is intended to
+ * search and link to a consortial catalog, such as ReShare, which uses its own VuFind
+ * instance to display consortium holdings and facilitate borrowing between institutions.
+ *
+ * @category VuFind
+ * @package  Recommendations
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+class ConsortialVuFind implements RecommendInterface, \Laminas\Log\LoggerAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+
+    /**
+     * Request parameter for the search query
+     *
+     * @var string
+     */
+    protected $requestParam = 'lookfor';
+
+    /**
+     * Number of results to show
+     *
+     * @var int
+     */
+    protected $limit = 5;
+
+    /**
+     * Connection to consortial VuFind API
+     *
+     * @var Connection
+     */
+    protected $connection;
+
+    /**
+     * ConsortialVuFind.ini configuration
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Base URL of a search results page
+     *
+     * @var string
+     */
+    protected $resultsBaseUrl = null;
+
+    /**
+     * Base URL of a record page
+     *
+     * @var string
+     */
+    protected $recordBaseUrl = null;
+
+    /**
+     * Any filters used in the search
+     *
+     * @var array
+     */
+    protected $searchFilters = [];
+
+    /**
+     * Boolean indicating if at least the minimal required configuration is present
+     *
+     * @var bool
+     */
+    protected $hasMinimumConfig = false;
+
+    /**
+     * Query string from the original search results
+     *
+     * @var string
+     */
+    protected $queryString;
+
+    /**
+     * Constructor
+     *
+     * @param Config     $config     ConsortialVuFind.ini configuration
+     * @param Connection $connection Connection to consortial VuFind API
+     */
+    public function __construct(
+        Config $config,
+        Connection $connection
+    ) {
+        $this->config = $config;
+        $this->connection = $connection;
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        $settings = explode(':', $settings);
+        $this->requestParam = !empty($settings[0]) ? $settings[0] : $this->requestParam;
+        $this->limit
+            = (isset($settings[1]) && is_numeric($settings[1]) && $settings[1] > 0)
```

Comment:
Could this be simplified to:

```
$limitSetting = intval($settings[1] ?? 0);
$this->limit = $limitSetting > 0 ? $limitSetting : $this->limit;
```

?

I realize this might cause some string inputs to get interpreted as 1... but I don't mind paying that price for more readable code.

Reply:
Sure, I made this change.  The original was copied from AbstractSearchObject.